### DISCUSSION TITLE
[Question] How to read from a client server's limits MaxMonitoredItemsPerSession and MaxSubscriptionsPerSession

### DISCUSSION THREAD

**Discussion by devaskim** [eclipse-milo/milo#1323] [General]
**Describe the bug**
Is there any API or some NodeIds to read values of MaxMonitoredItemsPerSession and MaxSubscriptionsPerSession limits from cliend side?

**Comment by kevinherron** [comment]
Look at the methods available on the ServerCapabilitiesType.

```
ServerTypeNode serverNode = (ServerTypeNode) client.getAddressSpace().getObjectNode(
    NodeIds.Server,
    NodeIds.ServerType
);

serverNode.getServerCapabilitiesNode().getMaxSubscriptionsPerSession();
serverNode.getServerCapabilitiesNode().getMaxMonitoredItemsPerSubscription();
```

`MaxMonitoredItemsPerSession` doesn't exist as a standard Node, though.

**Reply by devaskim** [reply]
> MaxMonitoredItemsPerSession doesn't exist as a standard Node, though.

Sorry, it was mistype - I meant `MaxMonitoredItemsPerSubscription`.
Thanks for help
